msticpy/data/drivers/github_data_types.py

```diff
@@ -0,0 +1,288 @@
+import attr
+from attr import attrs, attrib
+ 
+import requests
+import zipfile
+import io
+import re
+import glob
+import IPython
+import pandas as pd
+from ipywidgets import widgets as ipywidgets, Layout
+from IPython.display import display, HTML
+from pathlib import Path
+from requests.exceptions import HTTPError
+from typing import Callable, List, Optional, Set, Tuple, IO
+
+import yaml
+import json
+import csv
+import os
+
+import matplotlib.pyplot as plt
+import sys
+
+from pandas import json_normalize
+from tqdm.notebook import tqdm
+from datetime import date
+from datetime import datetime
+from requests_html import HTMLSession
+from collections import Counter
+
+@attrs
+class Query(object):
+    id: str = attrib(factory=str)
+    name: str = attrib(factory=str)
+    description: str = attrib(factory=str)
+    severity: str = attrib(factory=str)
+    tags: dict = attrib(factory=list)
+    requiredDataConnectors: dict = attrib(factory=dict)
+    queryFrequency: str = attrib(factory=str)
+    queryPeriod: str = attrib(factory=str)
+    triggerOperator: str = attrib(factory=str)
+    triggerThreshold: str = attrib(factory=str)
+    tactics: list = attrib(factory=list)
+    relevantTechniques: list = attrib(factory=list)
+    query: str = attrib(factory=list)
+    entityMappings: dict = attrib(factory=dict)
+    customDetails: dict = attrib(factory=dict)
+    alertDetailsOverride: dict = attrib(factory=dict)
+    version: str = attrib(factory=str)
+    kind: str = attrib(factory=str)
+
+def get_sentinel_queries_from_github(
+    git_url: Optional[
+        str
+    ] = "https://github.com/Azure/Azure-Sentinel/archive/master.zip",
+    outputdir: Optional[str] = None,
+) -> bool:
+    """
+    Download Microsoft Sentinel Github archive and extract detection and hunting queries.
+
+    Parameters
+    ----------
+    git_url : str, optional
+        URL of the GIT Repository to be downloaded, by default "https://github.com/Azure/Azure-Sentinel/archive/master.zip"
+    outputdir : str, optional
+        Provide absolute path to the output folder to save downloaded archive (e.g. '/usr/home' or 'C:\downloads'),
+        If no path provided, it will download to .msticpy dir under Azure-Sentinel directory.
+    """
+
+    if outputdir is None:
+        outputdir = Path.joinpath(Path("~").expanduser(), ".msticpy", "Azure-Sentinel")
+
+    try:
+        with requests.get(git_url, stream=True) as response:
```

Comment:
We've stopped using requests - use httpx instead - it's a drop-in replacement.
```
from ..common.pkg_config import get_http_timeout
from ..common.utility import mp_ua_header

....

response = httpx.get(
            submit_url, timeout=get_http_timeout(), headers=mp_ua_header()
        )
```

Reply:
httpx seems to get the state of the web page rather than downloading it - may be wrong and will look into this and update soon!